src/databricks/labs/ucx/workspace_access/groups.py

```diff
@@ -165,11 +165,22 @@ def prepare_groups_in_environment(self):
             logger.info("Using the provided group listing")
 
             for g in group_names:
-                assert g not in self.SYSTEM_GROUPS, f"Cannot migrate system group {g}"
-                assert self._get_group(g, "workspace"), f"Group {g} not found on the workspace level"
-                assert self._get_group(g, "account"), f"Group {g} not found on the account level"
-
-        if not group_names:
+                if g in self.SYSTEM_GROUPS:
+                    logger.info(f"Cannot migrate system group {g}. {g} will be skipped.")
+                    group_names.remove(g)
+                    continue
+                if not self._get_group(g, "workspace"):
+                    logger.info(f"Group {g} not found on the workspace level. {g} will be skipped.")
+                    group_names.remove(g)
```

Comment:
can you add a test to make sure that when the groups are either not found or invalid - they don't get to a selected list?